controllers/k8ssandra/cleanup.go

```diff
@@ -153,9 +153,144 @@ func (r *K8ssandraClusterReconciler) checkDcDeletion(ctx context.Context, kc *ap
 }
 
 func (r *K8ssandraClusterReconciler) deleteDc(ctx context.Context, kc *api.K8ssandraCluster, dcName string, logger logr.Logger) result.ReconcileResult {
+
+	// delete Stargate
+	if result := r.deleteStargate(ctx, kc, dcName, logger); result != nil {
+		return result
+	}
+
+	// delete Reaper
+	if result := r.deleteReaper(ctx, kc, dcName, logger); result != nil {
+		return result
+	}
+
+	// delete the DC
+	if result := r.deleteDatacenter(ctx, kc, dcName, logger); result != nil {
+		return result
+	}
+	delete(kc.Status.Datacenters, dcName)
+	logger.Info("DC deletion finished", "DC", dcName)
+	return result.Continue()
+}
+
+func (r *K8ssandraClusterReconciler) findStargateForDeletion(
+	ctx context.Context,
+	kcKey client.ObjectKey,
+	dcName string,
+	remoteClient client.Client) (*stargateapi.Stargate, error) {
+
+	if remoteClient == nil {
+		return nil, nil
+	}
+	stargateName := kcKey.Name + "-" + dcName + "-stargate"
+	selector := k8ssandralabels.PartOfLabels(kcKey)
+	options := &client.ListOptions{LabelSelector: labels.SelectorFromSet(selector)}
+	stargateList := &stargateapi.StargateList{}
+
+	err := remoteClient.List(ctx, stargateList, options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find Stargate (%s) for DC (%s) deletion: %v", stargateName, dcName, err)
+	}
+	for _, stargate := range stargateList.Items {
+		if stargate.Name == stargateName {
+			return &stargate, nil
+		}
+	}
```

Comment:
(Also applied to the original code) Can't we do a simple lookup by key with `remoteClient.Get()` instead of this?
We just need the namespace but it should be the same as `kcKey` if I'm not mistaken.